sdl_viewer/src/main.rs

```diff
@@ -98,17 +104,33 @@ struct NodeView {
     vertex_array: GlVertexArray,
     _buffer_position: GlBuffer,
     _buffer_color: GlBuffer,
+    _element_buffer: GlBuffer,
 }
 
 impl NodeView {
     fn new(program: &GlProgram, node_data: octree::NodeData) -> Self {
         let vertex_array = GlVertexArray::new();
         vertex_array.bind();
 
-        let buffer_position = GlBuffer::new();
-        let buffer_color = GlBuffer::new();
+        let buffer_position = GlBuffer::new(GlBufferKind::ArrayBuffer);
+        let buffer_color = GlBuffer::new(GlBufferKind::ArrayBuffer);
+        let element_buffer = GlBuffer::new(GlBufferKind::ElementArrayBuffer);
```

Comment:
Isn't this usually called an index buffer? And isn't this actually unnecessary here, since we could just reorder `buffer_position` and `buffer_color` according to the random shuffle `indices`, using `glDrawArrays` as before?